packages/router/router.ts

```diff
@@ -1514,7 +1516,7 @@ export function createRouter(init: RouterInit): Router {
 
     // Store off the match so we can call it's shouldRevalidate on subsequent
     // revalidations
-    fetchLoadMatches.set(key, [path, match, matches]);
+    fetchLoadMatches.set(key, [routeId, path, match, matches]);
```

Comment:
Track the calling `routeId` so we can skip `fetcher.load` revalidations on fetchers that are about to be deleted